### PR TITLE
OHRM5X-1453: Develop performance - final evaluation API

### PR DESCRIPTION
**Pull request by devishke-orange**
PR contains:
- OHRM5X-1453: [API] Final evaluation
- OHRM5X-1432: reopen fix

### CODE REVIEW

File: src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewFinalEvaluationAPI.php [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    public function getOne(): EndpointResult
+    {
+        $reviewId = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            self::PARAMETER_REVIEW_ID
+        );
+
+        $review = $this->getPerformanceReviewService()
+            ->getPerformanceReviewDao()
+            ->getPerformanceReviewById($reviewId);
```
Comment (SharukaNP) [inline src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewFinalEvaluationAPI.php:205]: Should we check for review status Id in here?

Comment (devishke-orange) [reply src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewFinalEvaluationAPI.php:205]:
```
$this->throwBadRequestExceptionIfStatusInArray(
            $review->getStatusId(),
            [PerformanceReview::STATUS_INACTIVE, PerformanceReview::STATUS_COMPLETED]
        );
```
throwing BadRequestException if status matches inactive or completed. For get one operation, only checking for inactive state

File: src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php [hunk 2]
```diff
@@ -416,7 +441,9 @@ public function getReviewIdsForSupervisorReviewer(int $supervisorEmpNumber): arr
      */
     public function getReviewIdList(): array
     {
+        // TODO check purged employee permissions
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php:444]: *

File: src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewFinalEvaluationAPI.php [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+                self::PARAMETER_FINAL_COMMENT,
+                new Rule(Rules::STRING_TYPE)
+            ),
+            new ParamRule(
+                self::PARAMETER_COMPLETED_DATE,
+                new Rule(Rules::API_DATE)
+            ),
+        ];
+
+        $complete = $this->getRequestParams()->getBoolean(
```
[the remark this answers is not in the document]
Comment (devishke-orange) [reply src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewFinalEvaluationAPI.php:152]: tried with null, string, number and validation works properly

**Pull request merged**